### DISCUSSION TITLE
React Server Component support

### DISCUSSION THREAD

**Discussion by AsuraKev** [web-infra-dev/modern.js#4653] [Q&A]
Are there plans to support RSC in the future?

**Comment by chenjiahan** [comment]
We will support the RSC when it becomes attractive enough.

Please refer to: https://modernjs.dev/en/guides/get-started/introduction.html#nextjs

> Modern.js considers both client-side rendering (CSR) and server-side rendering (SSR) to be equally important. When you build a Modern.js application, it defaults to client-side rendering, and you don't need to understand the uses and limitations of Server Components. You can also enable SSR whenever you need it, and even enable SSR for specific pages. The whole process is fully progressive. **Please note that Modern.js does not currently support Server Components. We will continue to monitor its potential and provide support when appropriate.**
